src/cedar.js

```diff
@@ -482,19 +482,20 @@ Cedar.prototype._renderSpec = function(spec){
     }
     //use vega to parse the spec
     //it will handle the spec as an object or url
-    vg.parse.spec(spec, function(chartCtor) {
+    vg.parse.spec(spec, function(error, chartCtor) {
+
+      if (error) { return error; }
 
       //create the view
       self._view = chartCtor({
         el: self._elementId,
         renderer: self._renderer
-      });
-
+      }).update();
 
       var width = self.width || parseInt(d3.select(self._elementId).style('width'), 10) || 500;
       var height = self.height || parseInt(d3.select(self._elementId).style('height'), 10) || 500;
```

Comment:
Also a note, I 'think' that 2) is the reason why #187 has that janky visualization if you don't set an appropriate height for the chart. I could be mistaken though. Regardless that one is occurring because the label itself is taking up the entire height of the chart element.